BinaryTree/BinaryTree.hpp

```diff
@@ -0,0 +1,99 @@
+/*  Binary Tree Implementation
+    Author: Andrew Burger */
+
+#ifndef ANDREW_CPP_LIBRARY_BINARYTREE_H__
+#define ANDREW_CPP_LIBRARY_BINARYTREE_H__
```

Comment:
I'd recommend you just use `#pragma once` unless you can't (e.g., you're using some kind of weird compiler).
Then you don't have to pick a symbol like `ANDREW_CPP_LIBRARY_BINARYTREE_H__` and hope it's actually unique.
It's also one line of code instead of three, one of which is far away from these two